common/djangoapps/third_party_auth/api/tests/test_views.py

```diff
@@ -130,3 +144,129 @@ def test_list_connected_providers__withapi_key(self, api_key, target_user, expec
         if expect_result == 200:
             self.assertIn("active", response.data)
             self.assertItemsEqual(response.data["active"], self.expected_active(target_user))
+
+
+@override_settings(EDX_API_KEY=VALID_API_KEY)
+@ddt.ddt
+@unittest.skipUnless(settings.ROOT_URLCONF == 'lms.urls', 'Test only valid in lms')
+class UserMappingViewAPITests(TpaAPITestCase):
+    """
+    Test the Third Party Auth User Mapping REST API
+    """
+    @ddt.data(
+        (VALID_API_KEY, PROVIDER_ID_TESTSHIB, 200, get_mapping_data_by_usernames(LINKED_USERS)),
+        ("i am an invalid key", PROVIDER_ID_TESTSHIB, 403, None),
+        (None, PROVIDER_ID_TESTSHIB, 403, None),
+        (VALID_API_KEY, 'non-existing-id', 404, []),
+    )
+    @ddt.unpack
+    def test_list_all_user_mappings_withapi_key(self, api_key, provider_id, expect_code, expect_data):
+        url = reverse('third_party_auth_user_mapping_api', kwargs={'provider_id': provider_id})
+        response = self.client.get(url, HTTP_X_EDX_API_KEY=api_key)
+        self._verify_response(response, expect_code, expect_data)
+
+    @ddt.data(
+        (PROVIDER_ID_TESTSHIB, 200, get_mapping_data_by_usernames(LINKED_USERS)),
+        ('non-existing-id', 404, []),
+    )
+    @ddt.unpack
+    def test_list_all_user_mappings_oauth2(self, provider_id, expect_code, expect_data):
+        url = reverse('third_party_auth_user_mapping_api', kwargs={'provider_id': provider_id})
+        # create oauth2 auth data
+        user = UserFactory.create(username='api_user')
+        client = Client.objects.create(name='oauth2_client', client_type=CONFIDENTIAL)
+        token = AccessToken.objects.create(user=user, client=client)
+        ProviderApiPermissions.objects.create(client=client, provider_id=provider_id)
```

Comment:
Can you add a negative test case here for OAuth2 to make sure the `ProviderApiPermissions` is actually enforcing when an unauthorized client accesses a provider?
Suggestion: You can add another `ddt` parameter for using an invalid access token.